crates/editor/src/element.rs

```diff
@@ -4938,6 +4994,26 @@ impl Element for EditorElement {
                         cx,
                     );
 
+                    self.editor.update(cx, |editor, cx| {
+                        let is_signature_help_displayed = editor.signature_help_state.is_some();
+                        if is_signature_help_displayed {
+                            editor.show_signature_help(&ShowSignatureHelp, cx);
+                            snapshot = editor.snapshot(cx);
```

Comment:
Should a snapshot be really updated after the signature showing? 
We have not done that for other invocations, and generally re-setting the state this way seems odd and dangerous — other snapshots may be around, unchanged.

Reply:
Come to think of this more, I do not think this `self.editor.update` block should be added at all here — what is says now, is that during every frame rendered (up to 120 fps), it spawns a task to query LSP and resolve things — sounds really bad.

Instead, we should follow that "is selection within brackets" approach as commented in the other note.

Reply:
@SomeoneToIgnore 
> Should a snapshot be really updated after the signature showing?

Thank you. I have removed the snapshot update as it is not necessary on dcf5f21cbb2cbf2a3817364834c6c2a73842ff6c

> Come to think of this more, I do not think this self.editor.update block should be added at all here — what it says now, is that during every frame rendered (up to 120 fps), it spawns a task to query LSP and resolve things — sounds really bad.

I fixed it on d2082de82 so that show_signature_help is called when the cursor position changes, rather than every time it's rendered.